web/src/composebox_typeahead.ts

```diff
@@ -674,9 +674,22 @@ export function get_person_suggestions(
     let groups: UserGroup[];
     if (opts.filter_groups_for_mention) {
         groups = user_groups.get_user_groups_allowed_to_mention();
-    } else if (opts.filter_groups_for_guests && !settings_data.user_can_access_all_other_users()) {
-        groups = user_groups.get_realm_user_groups().filter((group) => {
+    } else if (opts.filter_groups_for_dm) {
+        const can_access_all_users = settings_data.user_can_access_all_other_users();
+        groups = user_groups.get_all_realm_user_groups().filter((group) => {
             const group_members = group.members;
+            if (group_members.size > 20) {
+                // We do not want user trying to DM more than 20 users
+                // together.
+                return false;
+            }
```

Comment:
Hmm, is `group.members.size` the size of the group's direct members? I think we want to include recursive subgroup sizes here. 

This is a performance-sensitive code path, since it runs on typing every character. We should not walk the full graph of members using `get_recursive_group_members` to check if the group is big. I think a function that walks the group graph like `get_recursive_group_members` adding members to a `Set` and checking if it's at least a passed limit would work. So a `user_groups.is_group_smaller_than(group, max_size)`, would be fast and a good answer here.

Reply:
We might still need to call `get_recursive_group_members` for guests users to check if there is any group member which the guest user cannot access. So, we can call `get_recursive_group_members` only if user cannot access all other users.

Also, named the function to check number of members `is_group_larger_than` as we want the count of members to be less than or equal to 20.